optuna/logging.py

```diff
@@ -174,7 +174,50 @@ def enable_default_handler() -> None:
 def disable_propagation() -> None:
     """Disable propagation of the library log outputs.
 
-    Note that log propagation is disabled by default.
+    Note that log propagation is disabled by default. You only need to use this function
+    to stop log propagation when you use :func:`~optuna.logging.enable_propogation()`.
+
+    Example:
+
+        Stop propogating logs to the root logger and start logging to :obj:`sys.stderr`.
```

Comment:
A nit, but could you update the description of the example as well?
```suggestion
        Stop propagating logs to the root logger on the second optimize call.
```